game-core/src/main/java/org/triplea/game/server/AvailableGames.java

```diff
@@ -36,13 +35,23 @@
   private final Map<String, URI> availableGames;
 
   AvailableGames() {
-    availableGames =
-        FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
-            .parallelStream()
-            .map(AvailableGames::getGames)
-            .map(Map::entrySet)
-            .flatMap(Collection::stream)
-            .collect(Collectors.toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
+    availableGames = new HashMap<>();
+    FileUtils.listFiles(ClientFileSystemHelper.getUserMapsFolder())
+        .parallelStream()
+        .map(AvailableGames::getGames)
+        .map(Map::entrySet)
+        .flatMap(Collection::stream)
+        .forEach(
+            gameEntry -> {
+              if (!availableGames.containsKey(gameEntry.getKey())) {
+                availableGames.put(gameEntry.getKey(), gameEntry.getValue());
+              } else {
+                log.warning(
+                    String.format(
+                        "DUPLICATE GAME ENTRY! Ignoring game entry: %s, " + "existing value is: %s",
+                        gameEntry, availableGames.get(gameEntry.getKey())));
+              }
+            });
```

Comment:
While I don't think I can make a qualified decision if this kind of duplicate handling is the best approach to this, I want to point out that there's an overload of the `Collectors#toUnmodifianleMap` method that accepts a merge function:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collectors.html#toUnmodifiableMap(java.util.function.Function,java.util.function.Function,java.util.function.BinaryOperator)
The default implementation seems to be this one:

https://github.com/openjdk/jdk/blob/0a38584c10c1b0d971271f47934e4003bde7d986/src/java.base/share/classes/java/util/stream/Collectors.java#L148-L159

Which throws exactly the exception we are experiencing.
By providing this custom function we can keep the functional style + we maintain the unmodifiable map, which is no longer defensively copied or wrapped somehow for thread safety (the reason why unmodifiability would be an important thing to keep)

Reply:
The merge function looks really attractive. I just gave it a try, but it looks like we won't have access to the 'key' in conflict.

The merge function essentially allows `(oldValue, newValue) -> deconflictedValue`. While we can figure out the conflicting game name based on the URI alone, it certainly is more direct if we can log the game key that has the duplicate value.

Reply:
Ah, that's something I didn't think of. Unfortunate.

Even though this PR is already merged I have some concerns.
This is a parallel Stream, so we could end up running into concurrency issues because the now unsynchronized map doesn't have a wrapper for this kind of protection.
Also somewhere down the line we give this exact map away without any sort of write protection, which might introduce weird bugs if we modify it somewhere by accident.

I was able to come up with 4 potential solutions to this:
1. Just wrap it in a Collections.synchronizedMap package and return an unmodifiable collection wrapper int the API (I forgot where this was, but I removed it in my PR), the easiest solution but might hurt parallel performance and adds unecessary abstraction.
2. Do it in 2 steps. We could use the streams and everything to put all of the entry objects in an unmodifiable List and use a normal for-each loop to fill a map, and all we store is the unmodifiable wrapper to this map. Easy solution, but still not optimal use of streams IMO.
3. Same as 2. but instead of using for-each we use the same collector as before and iterate over all elements until we find the correct key in case we find a conflicting value. Still weird, but works as well.
4. Provide a custom collector. Instead of using the default `Collectors#toUnmodifiableMap` we could provide our own "implementation". The `Collector` interface offers some handy static functions for this.
This would add some complexity:
```java
Collector.of(
  HashMap::new,
  (map, entry) -> {
    // Alternative approach would be to use Map#containsKey
    if (map.put(entry.getKey(), entry.getValue()) != null) {
      log.log(Level.INFO, "Duplicate key: " + entry.getKey());
    }
  },
  (map1, map2) -> {
    for (final var entry : m2.entrySet()) {
      // Variant two, as described earlier
      if (map1.containsKey(entry.getKey())) {
        log.log(Level.INFO, "Duplicate key: " + entry.getKey());
      }
      map1.put(entry.getKey(), entry.getValue())
    }
    return m1;
  },
  Collections::unmodifiableMap,
  // Optional but for the sake of completeness
  Collector.Characteristics.UNORDERED);
```

The only thing I don't like about approach 4 is that it duplicates the detection logic, but this could be solved by extracting it to a private static helper method, so it would look like this:
```java
Collector.of(
  HashMap::new,
  (map, entry) -> putAndWarn(map, entry.getKey(), entry.getValue()),
  (map1, map2) -> {
    for (final var entry : map2.entrySet()) {
      putAndWarn(map1, entry.getKey(), entry.getValue());
    }
    return map1;
  },
  Collections::unmodifiableMap,
  // Optional but for the sake of completeness
  Collector.Characteristics.UNORDERED);
```

What do you think? I leaning towards option 4 simply because I think it's the most consistent way to do it.